pkg/pool-weighted/test/ManagedPool.test.ts

```diff
@@ -100,222 +133,177 @@ describe('ManagedPool', function () {
-        const startingBpt = await pool.balanceOf(owner);
+      context('when LP is on the allowlist', () => {
+        sharedBeforeEach('add address to allowlist', async () => {
+          await pool.addAllowedAddress(owner, other.address);
+        });
 
-        await expect(pool.joinAllGivenOut({ from: owner, bptOut: startingBpt })).to.be.revertedWith(
-          'ADDRESS_NOT_ALLOWLISTED'
-        );
-      });
-    });
+        it('the listed LP can join', async () => {
+          await pool.joinAllGivenOut({ from: other, bptOut: FP_ONE });
 
-    context('when mustAllowlistLPs is toggled', () => {
-      sharedBeforeEach('initialize pool', async () => {
-        await pool.addAllowedAddress(owner, owner);
-        await pool.init({ from: owner, initialBalances });
+          expect(await pool.balanceOf(other)).to.be.gt(0);
+        });
       });
 
-      it('allows owner to turn it off (open to public LPs)', async () => {
-        const startingBpt = await pool.balanceOf(owner);
-
-        const receipt = await pool.setMustAllowlistLPs(owner, false);
-        expectEvent.inReceipt(await receipt.wait(), 'MustAllowlistLPsSet', {
-          mustAllowlistLPs: false,
+      context('when LP is not on the allowlist', () => {
+        it('it reverts', async () => {
+          await expect(pool.joinAllGivenOut({ from: other, bptOut: FP_ONE })).to.be.revertedWith(
+            'ADDRESS_NOT_ALLOWLISTED'
+          );
         });
-
-        // Should be turned off
-        expect(await pool.getMustAllowlistLPs()).to.be.false;
-
-        // And allow joins from anywhere
-        await expect(pool.joinAllGivenOut({ from: other, bptOut: startingBpt })).to.not.be.reverted;
-
-        // Does not allow adding or removing addresses now
-        await expect(pool.addAllowedAddress(owner, other.address)).to.be.revertedWith('FEATURE_DISABLED');
-        await expect(pool.removeAllowedAddress(owner, other.address)).to.be.revertedWith('FEATURE_DISABLED');
       });
     });
-  });
 
-  describe('with valid creation parameters', () => {
-    context('when initialized with swaps disabled', () => {
+    context('when swaps are disabled', () => {
       sharedBeforeEach('deploy pool', async () => {
         const params = {
           tokens: poolTokens,
           weights: poolWeights,
-          owner: owner.address,
           poolType: WeightedPoolType.MANAGED_POOL,
-          swapEnabledOnStart: false,
+          vault,
+          swapEnabledOnStart: true,
+          owner: owner.address,
         };
         pool = await WeightedPool.create(params);
+        await pool.init({ from: other, initialBalances });
       });
 
-      it('swaps are blocked', async () => {
-        await expect(pool.swapGivenIn({ in: 1, out: 0, amount: fp(0.1) })).to.be.revertedWith('SWAPS_DISABLED');
+      sharedBeforeEach('disable swaps', async () => {
+        await pool.setSwapEnabled(owner, false);
+      });
+
+      context('proportional joins/exits', () => {
```

Comment:
```suggestion
      context('proportional joins', () => {
```

We're testing joins in this describe block